services/head_tracker.go

```diff
@@ -63,28 +65,32 @@ func (ht *HeadTracker) Start() error {
 		ht.number = &numbers[0]
 	}
 
-	ht.headers = make(chan models.BlockHeader)
```

Comment:
If we `Start`, `Stop`, then `Start` again. We'll be in a bad place with regard to `ht.headers`. The `close` signals the goroutine to close down. Perhaps that's a scenario that simply doesn't matter in practice.